client/blocks/stats-sparkline/index.jsx

```diff
@@ -4,56 +4,72 @@
 
 import PropTypes from 'prop-types';
 import React, { Fragment } from 'react';
-import { connect } from 'react-redux';
-import { get, mapValues, sortBy } from 'lodash';
+import classnames from 'classnames';
+import { useSelector } from 'react-redux';
 import { useTranslate } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
 import QuerySiteStats from 'calypso/components/data/query-site-stats';
-import { isJetpackSite, getSiteOption } from 'calypso/state/sites/selectors';
 import { getSiteStatsNormalizedData } from 'calypso/state/stats/lists/selectors';
 
-const StatsSparkline = ( { isJetpack, siteUrl, className, siteId, highestViews } ) => {
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const StatsSparklineChart = ( { className, hourlyViews } ) => {
 	const translate = useTranslate();
+	const highestViews = Math.max( ...hourlyViews );
+	const title = translate( 'Highest hourly views %(highestViews)s', { args: { highestViews } } );
 
-	if ( ! siteId || ! siteUrl || isJetpack ) {
-		return null;
-	}
+	const chartHeight = 24 - 7; // 24px is the desired height, 7px is the total top+bottom padding
+	const chartWidth = 2 * hourlyViews.length - 1; // 1px bars with 1px space in between
 
-	const title = highestViews
-		? translate( 'Highest hourly views %(highestViews)s', { args: { highestViews } } )
-		: null;
+	return (
+		<div
+			className={ classnames( 'stats-sparkline', className ) }
+			title={ title }
+			style={ { height: chartHeight + 'px', width: chartWidth + 'px' } }
+		>
+			{ hourlyViews.map( ( value, i ) => {
+				// for zero value, we show a baseline bar with 1px height
+				let scale = 1 / chartHeight;
+				// if the chart is all zeros, show just the flat baseline
+				if ( highestViews > 0 ) {
+					// fill the remaining height with the bar scaled according to the value
+					scale += ( value / highestViews ) * ( 1 - 1 / chartHeight );
+				}
 
-	const src = `${ siteUrl }/wp-includes/charts/admin-bar-hours-scale-2x.php?masterbar=1&s=${ siteId }`;
+				return (
+					<div
+						key={ i }
```

Comment:
Nit: should we use some sort of a string prefix before the `i`? Not that it matters so much, but from a strict type point of view, `i` is a number, and react keys are supposed to be strings. By changing `i` to `sparkline-bar-${ i }` or something will also implicitly convert it to string.

Reply:
Using the numeric index as `key` is pretty common in cases where a better key is not available, and `@types/react` define the `Key` type as `string | number`. So a simple `key={ i }` seems fine.

No prefix is needed even to make the keys unique, although folks often add it just to be sure. The keys are scoped locally for the JSX container and never leak outside.

Reply:
Exactly. Numbers are problematic in DOM `id` (you can't querySelect them easily). But in React keys they're fine and aren't global.